### ISSUE TITLE
Can we have lower bound generic type constraint?

### ISSUE THREAD

**Issue by voroninp** [dotnet/coreclr#20505]
Unfortunately, now it is not possible to write this code:

```CSharp
interface Foo<out T>
{
     TNew Change<TNew>(TNew item) where T: TNew
}
```

JVM has this [feature](https://docs.oracle.com/javase/tutorial/java/generics/lowerBounded.html) and it is extensively used in Scala when working with [variance](https://docs.scala-lang.org/tour/lower-type-bounds.html). 

How hard is it to add this kind of constraints?

**Comment by svick** [comment]
I think this is the same proposal as https://github.com/dotnet/csharplang/issues/1621.

**Comment by voroninp** [comment]
@svick Yes, in essence they are the same, but I did not expect to find the the proposal among C# issues because it is not supported by CLR, so I didn't even search there.
